### PR TITLE
Canal starts sync by GTID

### PR DESCRIPTION
**Pull request by NearTan**
https://github.com/siddontang/go-mysql/issues/144
- Add a Canal entry function for GTID
- Provide GTID in `masterInfo`
- Handle GTID events
- Fix typo `writeBinglogDumpCommand` -> `writeBinlogDumpCommand`

### CODE REVIEW

File: canal/sync.go [hunk 1]
```diff
@@ -3,26 +3,44 @@ package canal
...
+	"github.com/satori/go.uuid"
 )
 
 var (
 	expAlterTable = regexp.MustCompile("(?i)^ALTER\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
 )
 
-func (c *Canal) startSyncBinlog() error {
-	pos := c.master.Position()
+func getSyncer(c *Canal) (*replication.BinlogStreamer, error) {
```
Comment (siddontang) [inline canal/sync.go]: getSyncer here is not proper. This function seems to start the sync. 

you can name to `(c *Canal) startSyncer()` and then rename latter `startSyncBinlog` to `runSyncBinlog`

**Comment by siddontang** [comment]
Thanks @NearTan

File: canal/sync.go (hunk 1 above)
Comment (NearTan) [inline canal/sync.go]: I think that rename `startSyncBinlog` to `runSyncBinlog` will cause compatibility issues. 

What about delete `getSyncer` function and implement function logic in `startSyncBinlog`?

Comment (siddontang) [inline canal/sync.go]: @NearTan 

`startSyncBinlog` is not exported, so it can't cause the compatibility issue.

Comment (NearTan) [inline canal/sync.go]: @siddontang fixed, plz review again

**Comment by siddontang** [comment]
Thanks @NearTan

**Pull request merged**